Quick_Deploy/vLLM/model_repository/vllm/1/model.py

```diff
@@ -0,0 +1,290 @@
+        """
+        # first await the shutdown signal
+        while self._shutdown_event.is_set() is False:
+            await asyncio.sleep(5)
+
+        # then wait for all in-flight tasks to complete
+        while self._inflight_counter > 0:
+            self.logger.log_info("Awaiting remaining {} inflight requests".format(self._inflight_counter))
+            await asyncio.sleep(5)
+
+        self.logger.log_info("Shutdown complete")
+
+    def make_sample_params(self, request: CompletionRequest) -> SamplingParams:
+        """
+        Creates a vLLM SamplingParams object from the given arguments.
+        """
+        # todo(ryan): make more defaults configurable via the intialized args
+        return SamplingParams(
+            n=1,
+            best_of=1,
+            temperature=request.temperature,
+            top_p=request.top_p,
+            top_k=request.top_k,
+            stop=request.stop,
+            ignore_eos=request.ignore_eos,
+            max_tokens=request.max_tokens,
+        )
+
+    def make_output_options(self, req: CompletionRequest) -> OutputOptions:
+        """
+        Creates an OutputOptions object from the given arguments.
+        """
+        # todo(ryan): make more defaults configurable via the intialized args
+        return OutputOptions(
+            echo=req.echo,
+            stream=req.stream,
+        )
+
+    def preprocess(self, request) -> Tuple[str, SamplingParams, OutputOptions]:
+        """
+        Converts a Triton request into a VLLM input by extracting the prompt and sampling parameters.
+        """
+        # deserialize the VLLM request from the Triton InferRequest
+        try:
+            serialized_json = (
+                pb_utils.get_input_tensor_by_name(request, "serialized_request_json").as_numpy()[0]
+            )
+            request = CompletionRequest(**json.loads(serialized_json))
+
+            return (
+                request.prompt,
+                self.make_sample_params(request),
+                self.make_output_options(request),
+            )
+        except Exception as e:
+            self.logger.log_info(f"Error parsing request: {e}")
+            raise e
+
+    def postprocess(self, output: VLLMOutput, output_options: OutputOptions, inflight_avg: float):
+        """
+        Converts a VLLM output into a Triton response.
+        """
+        # create a Triton Response from the VLLM output
+        response = CompletionResponse()
+        response.request_id = output.request_id
+        response.finished = output.finished
+
+        # number of tokens in the prompt and total number of tokens generated across all completions
+        response.prompt_token_count = len(output.prompt_token_ids)
+
+        # optionally echo the prompt back to the client
+        if output_options.echo:
+            response.prompt = output.prompt
+            # response.prompt_token_ids = output.prompt_token_ids
```

Comment:
Do we want to remove this line as we don't have `prompt_token_ids` field for CompletionResponse?